internal/pkg/externalplugins/blunderbuss/blunderbuss.go

```diff
@@ -107,14 +115,46 @@ func HandlePullRequestEvent(gc githubClient, pe *github.PullRequestEvent,
 	repo := &pe.Repo
 	opts := cfg.BlunderbussFor(repo.Owner.Login, repo.Name)
 
-	// Wait a few seconds to allow other automation plugin to apply labels.
-	gracePeriod := DefaultGracePeriodDuration * time.Second
+	// Wait a few seconds to allow other automation plugin to apply labels (Mainly sig label).
+	gracePeriod := time.Duration(opts.GracePeriodDuration) * time.Second
+	sleep(gracePeriod)
 
-	if opts.GracePeriodDuration != 0 {
-		gracePeriod = time.Duration(opts.GracePeriodDuration) * time.Second
+	// Check if PR has sig label.
+	if opts.RequireSigLabel && !isPRContainSigLabel(pr) {
```

Comment:
I think we can check this before sleep.

Reply:
And better test this logic.

Reply:
> I think we can check this before sleep.

I don't think so. In the repo required sig label, when a PR opened, it will not have a label, so it is meaningless to judge whether it contains the sig label.

Another problem is that we still need to distinguish whether the tag is added by robot or by human, otherwise when creating a PR, request view will be triggered twice (Triggered by pull request opened and label added events)